gnocchi/rest/aggregates/api.py

```diff
@@ -313,27 +297,46 @@ def groupper(r):
 
             return response
 
+    def get_measures_grouping(self, body, details, fill, granularity,
+                              needed_overlap, references, resources, start,
+                              stop, groupby):
+        def groupper(r):
+            return tuple((attr, r[attr]) for attr in groupby)
+
+        results = []
+        for key, resources in itertools.groupby(resources, groupper):
+            try:
+                results.append({
+                    "group": dict(key),
+                    "measures": self._get_measures_by_name(
+                        resources, references, body["operations"],
+                        start, stop, granularity, needed_overlap, fill,
+                        details=details)
+                })
+            except indexer.NoSuchMetric:
+                pass
+        return results
+
+    def get_measures_grouping_with_history(self, attr_filter, body, details,
+                                           fill, granularity, groupby,
+                                           needed_overlap, references,
+                                           sorts, start, stop):
+        results = []
+        try:
+            grouper = measures_grouper.Grouper(groupby, start, stop, body,
+                                               sorts, attr_filter, references,
+                                               granularity, needed_overlap,
+                                               fill, details)
+            results = grouper.get_grouped_measures()
+        except indexer.NoSuchMetric:
+            pass
+        return results
+
     @staticmethod
     def _get_measures_by_name(resources, metric_wildcards, operations,
```

Comment:
Including the exception however could include information about internals that should not be exposed, so either we should just leave it and give a 500 error and treat it as a bug, or return 400 and hope that it's valid. I'm more for 500 than returning 400 and it's potentially not an input issue.

Reply:
Yes, but in that case, we have some problems to discuss here:
 - About line 252
   - If we catch the Exception and abort it with an error 500 or 400, in some cases the Exception is an instance of `weboc.HTTPExceptions` that is some already handled exception, so in the case this Exception raises an HTTP 400 and we catch it to an HTTP 500, it will be an inconsistency. To try to solve it, we could catch an `weboc.HTTPExceptions` before catching the Exception itself and and ignore it.
   - If we dont catch the generic Exception, we will delegate to who called this method to handle it, and its ok in my opinion, and that is what we are doing here right now, except that we are logging the Exception before throwing it. This is the initial behavior of this procedure (we only added the logging stuff).
 - About line 331. No reason, I missed this one, sorry.